.github/workflows/maven.yml

```diff
@@ -39,6 +39,6 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-maven-
     - name: Build with Maven
-      run: ./mvnw -Pjakarta-apis -V -B -ff -s .github/settings.xml install '-Dmaven.test.skip=true' '-Dmaven.javadoc.skip=true'
+      run: ./mvnw -Pjakarta-apis -V -B -ff -s .github/settings.xml install '-DskipTests=true' '-Dmaven.javadoc.skip=true'
```

Comment:
Running with `maven.test.skip` not only skips tests it doesn't even compile them (which means any dependencies required at compile scope for tests will not be detected). maven-dependency-plugin requires all of the code to be compiled as it performs bytecode analysis to determine what is needed.